engine/table/src/main/java/io/deephaven/engine/util/AbstractScriptSession.java

```diff
@@ -61,7 +60,7 @@ protected AbstractScriptSession(@Nullable Listener changeListener, boolean isDef
         this.changeListener = changeListener;
 
         final UUID scriptCacheId = UuidCreator.getRandomBased();
-        classCacheDirectory = new File(CLASS_CACHE_LOCATION, UuidCreator.toString(scriptCacheId));
+        classCacheDirectory = CLASS_CACHE_LOCATION.resolve(UuidCreator.toString(scriptCacheId)).toFile();
```

Comment:
We discussed introducing props:
INSTANCE_ID (if set, this is the ID)
INSTANCE_ID_FILE (if set and INSTANCE_ID isn't this is the file we load from or generate)
else, we randomly allocate an instance ID and be sure to clean up on exit.

Reply:
Punting - https://github.com/deephaven/deephaven-core/issues/1713